### PR TITLE
Deprecate shipping of component code in profiles

### PR DESCRIPTION
**Pull request by jrha**
Related to quattor/configuration-modules-core#553

### CODE REVIEW

File: quattor/types/component.pan [hunk 1]
```diff
@@ -18,6 +18,6 @@ type structure_component = extensible {
     "version"           ? string with match(SELF, '^\d+\.\d+\.\d+$')
     "register_change"   ? element*[]
     "dependencies"      ? structure_component_dependency
-    "code"              ? structure_component_code
+    "code"              ? structure_component_code with deprecated(0, "Shipping component code in the profiles is deprecated and is non-functional.")
```
Comment (stdweird) [inline quattor/types/component.pan]: should return `true` like 
https://github.com/quattor/configuration-modules-core/blob/master/ncm-metaconfig/src/main/pan/components/metaconfig/schema.pan#L18

File: quattor/types/component.pan [hunk 2]
```diff
@@ -18,6 +18,6 @@ type structure_component = extensible {
     "version"           ? string with match(SELF, '^\d+\.\d+\.\d+$')
     "register_change"   ? element*[]
     "dependencies"      ? structure_component_dependency
-    "code"              ? structure_component_code
+    "code"              ? structure_component_code with { deprecated(0, "daemon property has been deprecated, daemons should be used instead"); true; }
```
Comment (ned21) [inline quattor/types/component.pan]: s/daemon/code/

**Comment by jrha** [comment]
Unit tests will fail until other problems in this repo are fixed and not before RC1.

**Comment by stdweird** [comment]
@jrha don't know if we agreed to anything wrt that yesterday: should we already merge or wait for rc1 without template-library-core?

**Comment by stdweird** [comment]
hmm, but you merged #77. so i guess it's ok

**Comment by jrha** [comment]
Don't wait, just merge.

**Pull request merged**